libs/zzz/formula-ui/src/components/TagDisplay.tsx

```diff
@@ -1,11 +1,49 @@
+import { iconInlineProps } from '@genshin-optimizer/common/svgicons'
 import { ColorText } from '@genshin-optimizer/common/ui'
+import type { StatKey } from '@genshin-optimizer/zzz/consts'
+import { statKeyTextMap } from '@genshin-optimizer/zzz/consts'
 import type { Tag } from '@genshin-optimizer/zzz/formula'
+import { StatIcon } from '@genshin-optimizer/zzz/svgicons'
+import { StatDisplay } from '@genshin-optimizer/zzz/ui'
 import { getVariant, tagFieldMap } from '../char'
 import { getTagLabel } from '../util'
 export function TagDisplay({ tag }: { tag: Tag }) {
   return (
     <ColorText color={getVariant(tag)}>
-      {tagFieldMap.subset(tag)[0]?.title || getTagLabel(tag)}
+      <TagStrDisplay tag={tag} />
     </ColorText>
   )
 }
+const extraHandlingStats = ['hp', 'hp_', 'atk', 'atk_', 'def', 'def_'] as const
+const isExtraHandlingStats = (
+  stat: string
+): stat is (typeof extraHandlingStats)[number] =>
+  extraHandlingStats.includes(
+    stat as 'hp' | 'hp_' | 'atk' | 'atk_' | 'def' | 'def_'
+  )
+const qtMap = {
+  initial: 'Initial',
+  combat: 'Combat',
+  final: 'Final',
+  base: 'Base',
+}
+function TagStrDisplay({ tag }: { tag: Tag }) {
```

Comment:
Seems like this logic should be inside getTagLabel?

Reply:
If not, should clean up todo in getTagLabel, tho I wouldn't know what the purpose of that function is other than what this function does

Reply:
probably wait for next refactor with SRO